### ISSUE TITLE
feat(docs): add advanced user guide for custom mixin

### ISSUE THREAD

**Issue by kemingy** [mosecorg/mosec#385]
### Describe the feature

We already have several mixin implementations. It's necessary to have a development tutorial for users who need to custom their own mixins.

**Comment by Keerthi54654** [comment]
@kemingy - 
Explore Existing Mixins: Review current mixin implementations in the repository.
Testing Framework: Utilize available testing tools to validate the functionality of the mixins.
Documentation: Present the findings in a structured manner, highlighting best practices for creating and using mixins effectively.
Is this how u want it to be there

**Comment by kemingy** [comment]
> @kemingy - Explore Existing Mixins: Review current mixin implementations in the repository. Testing Framework: Utilize available testing tools to validate the functionality of the mixins. Documentation: Present the findings in a structured manner, highlighting best practices for creating and using mixins effectively. Is this how u want it to be there

No. I mean a complementary guide about how to write a custom Mixin for the advanced user. This should be a middle-level task.
